milabench/metadata.py

```diff
@@ -0,0 +1,42 @@
+import os
+from datetime import datetime
+import subprocess
+import cpuinfo
+
+from voir.instruments.gpu import get_gpu_info
+
+
+def _exec(cmd, default):
+    try:
+        return subprocess.check_output(cmd.split(" "), encoding="utf-8").strip()
+    except:
```

Comment:
We should log the exception somewhere.

Also there should be a validator that checks that the tag/commit metadata is present, because we should require it for submissions.

Reply:
I added a warning that version could not be retrieved.
and implemented a validation layer to make sure version was passed